junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ExtensionRegistry.java

```diff
@@ -88,28 +90,47 @@ private ExtensionRegistry(ExtensionRegistry parent) {
 	}
 
 	/**
-	 * Get all {@code Extensions} in this registry or one of its ancestors of
+	 * Stream all {@code Extensions} in this registry or one of its ancestors of
 	 * the specified type.
 	 *
 	 * @param extensionType the type of {@link Extension} to get
 	 * @see #getReversedExtensions(Class)
-	 * @see #stream(Class)
+	 * @see #getExtensions(Class)
 	 */
-	public <E extends Extension> List<E> getExtensions(Class<E> extensionType) {
-		List<E> extensions = new ArrayList<>();
-
-		if (this.parent != null) {
-			extensions.addAll(this.parent.getExtensions(extensionType));
+	public <E extends Extension> Stream<E> streamExtensions(Class<E> extensionType) {
```

Comment:
I actually prefer verbs for methods/functions - I'd maybe use nouns for properties if I ditch the get/set. It could be argued that this is such a case. But as you guessed, I wanted to keep this close to `getExtensions` and `getExtensionsAsStream` does that even better so I'm open to that as well. :smiley: